Lib/multiprocessing/spawn.py

```diff
@@ -40,7 +40,10 @@
 
 def set_executable(exe):
     global _python_exe
-    _python_exe = exe
+    if sys.platform == 'win32':
+        _python_exe = os.fsdecode(exe)
+    else:
+        _python_exe = os.fsencode(exe)
```

Comment:
`_python_exe` should not be assigned directly on lines 36-39.

Reply:
Oh sorry it was already in your suggested code but I did not pay enough attention. You are absolutely right, let’s fix that information leak too. Thanks!